src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -279,38 +282,51 @@ private static void RecordRequestStartEventLog(HttpContext httpContext)
             {
                 return null;
             }
-
             var headers = httpContext.Request.Headers;
-            var requestId = headers.TraceParent;
-            if (requestId.Count == 0)
-            {
-                requestId = headers.RequestId;
-            }
+            _propagator.ExtractTraceIdAndState(headers,
+                static (object? carrier, string fieldName, out string? fieldValue, out IEnumerable<string>? fieldValues) =>
+                {
+                    fieldValue = default;
+                    fieldValues = default;
+                    if (carrier is null)
```

Comment:
> if (carrier is null) [](http://example.com/codeflow?start=19&length=21)

is this possible in yor case?

Reply:
Not that I'm aware. I did for the nullability warnings I got otherwise

Reply:
you can use `carrier!` to suppress that.

Reply:
Done